selfservice/flow/login/organizations_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2025 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package login
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ory/kratos/identity"
+	"github.com/ory/kratos/session"
+)
+
+func TestPrepareOrganizations(t *testing.T) {
+	t.Run("should return empty filter if AAL is not AAL1", func(t *testing.T) {
+		f := &Flow{RequestedAAL: identity.AuthenticatorAssuranceLevel2}
+		r := &http.Request{URL: new(url.URL)}
+		sess := &session.Session{}
+
+		filters := PrepareOrganizations(r, f, sess)
+		assert.Empty(t, filters)
+	})
+
+	t.Run("should return empty filter if OrganizationID is not valid", func(t *testing.T) {
+		f := &Flow{RequestedAAL: identity.AuthenticatorAssuranceLevel1}
+		r := &http.Request{URL: new(url.URL)}
+		sess := &session.Session{}
+
+		filters := PrepareOrganizations(r, f, sess)
+		assert.Empty(t, filters)
+	})
+
+	t.Run("should return organization filter if OrganizationID is valid", func(t *testing.T) {
+		f := &Flow{RequestedAAL: identity.AuthenticatorAssuranceLevel1}
+		r := &http.Request{URL: new(url.URL)}
+		sess := &session.Session{
+			Identity: &identity.Identity{OrganizationID: uuid.NullUUID{Valid: true}},
+		}
+
+		filters := PrepareOrganizations(r, f, sess)
+		require.NotEmpty(t, filters)
+		assert.Equal(t, organizationFilter, filters)
+	})
+
+	t.Run("should parse OrganizationID from URL query if not in session", func(t *testing.T) {
+		f := &Flow{RequestedAAL: identity.AuthenticatorAssuranceLevel1}
+		r := &http.Request{
+			URL: &url.URL{
+				RawQuery: "organization=123e4567-e89b-12d3-a456-426614174000",
+			},
+		}
+		sess := &session.Session{}
+		filters := PrepareOrganizations(r, f, sess)
+		require.NotEmpty(t, filters)
+		assert.Equal(t, organizationFilter, filters)
+	})
```

Comment:
Missing cases:
- organization ID in flow
- organization ID in query and identity (matching and non-matching)
- organization ID set but AAL2